src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -176,9 +217,470 @@ private Object createObject(String className) {
             declaredConstructor.setAccessible(true);
             instance = declaredConstructor.newInstance();
         }
-        catch (final ReflectiveOperationException | NoClassDefFoundError exception) {
-            LOG.debug(IGNORING_EXCEPTION_MESSAGE, exception);
+        catch (final ReflectiveOperationException | NoClassDefFoundError ignored) {
+            // keep looking, ignoring exception
         }
         return instance;
     }
+
+    /**
+     * Fill short-to-full module names map.
+     */
+    private static void fillShortToFullModuleNamesMap() {
+        fillChecksFromAnnotationPackage();
+        fillChecksFromBlocksPackage();
+        fillChecksFromCodingPackage();
+        fillChecksFromDesingPackage();
+        fillChecksFromHeaderPackage();
+        fillChecksFromImportsPackage();
+        fillChecksFromIndentationPackage();
+        fillChecksFromJavadocPackage();
+        fillChecksFromMetricsPackage();
+        fillChecksFromModifierPackage();
+        fillChecksFromNamingPackage();
+        fillChecksFromRegexpPackage();
+        fillChecksFromSizesPackage();
+        fillChecksFromWhitespacePackage();
+        fillModulesFromChecksPackage();
+        fillModulesFromFilefiltersPackage();
+        fillModulesFromFiltersPackage();
+        fillModulesFromCheckstylePackage();
+    }
+
+    /**
+     * Fill short-to-full module names map with Checks from annotation package.
+     */
+    private static void fillChecksFromAnnotationPackage() {
+        NAME_TO_FULL_MODULE_NAME.put("AnnotationLocationCheck",
+                "com.puppycrawl.tools.checkstyle.checks.annotation.AnnotationLocationCheck");
```

Comment:
Some canonical names do not fit in a single line, so they need to be split - that is where the first part is the same String literal. If name fits line, then `BASE_PACKAGE` is not used to avoid creating extra Strings.